scrunch/datasets.py

```diff
@@ -1441,12 +1481,34 @@ class Variable(object):
 
     CATEGORICAL_TYPES = {'categorical', 'multiple_response', 'categorical_array'}
 
-    def __init__(self, resource, dataset_resource):
-        self.resource = resource
-        self.url = self.resource.self
-        self.dataset = Dataset(dataset_resource)
+    def __init__(self, var_shoji, dataset):
+        self.shoji_tuple = var_shoji
+        self.is_instance = False
+        self._resource = None
+        self.url = None
+        # pass an Entity class to save time
+        if isinstance(dataset, pycrunch.shoji.Entity):
+            self.dataset = dataset
+        else:
+            self.dataset = Dataset(dataset)
```

Comment:
Why does the variable need a `Dataset` instance inside it? It should be able to make do with just the Shoji Entity, the `Dataset` instance should have helper methods that a scrunch user needs, but I don't think any operation that we need internally in the Variable instance, or do we? 

We'll be able to tell if all our usages are `self.dataset.resource`, then we don't need the instance.